File: competitive_programming/leetcode/07_typescript/26_remove_duplicates_sorted_array.ts
function removeDuplicates(nums: number[]): number[] {
  const result: number[] = [];
  for (const value of nums) if (result.length === 0 || result[result.length - 1] !== value) result.push(value);
  return result;
}

if (removeDuplicates([1, 1, 2]).join(",") !== "1,2") throw new Error("remove duplicates failed");
if (removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]).join(",") !== "0,1,2,3,4") throw new Error("remove duplicates failed");
console.log("26 remove duplicates ok");

File: competitive_programming/leetcode/11_golang/26_remove_duplicates_sorted_array.go
package main

import "fmt"

func removeDuplicates(nums []int) []int {
	result := []int{}
	for _, value := range nums {
		if len(result) == 0 || result[len(result)-1] != value {
			result = append(result, value)
		}
	}
	return result
}

func main() {
	if fmt.Sprint(removeDuplicates([]int{1, 1, 2})) != "[1 2]" {
		panic("remove duplicates failed")
	}
	if fmt.Sprint(removeDuplicates([]int{0, 0, 1, 1, 1, 2, 2, 3, 3, 4})) != "[0 1 2 3 4]" {
		panic("remove duplicates failed")
	}
	fmt.Println("26 remove duplicates ok")
}
